Orange/canvas/application/errorreporting.py

```diff
@@ -0,0 +1,207 @@
+import os
+import sys
+import time
+import logging
+import platform
+import traceback
+import uuid
+from html import escape
+from threading import Thread
+
+from tempfile import mkstemp
+from collections import OrderedDict
+from urllib.parse import urljoin, urlencode
+from urllib.request import pathname2url, urlopen
+from unittest.mock import patch
+
+from PyQt4.QtCore import pyqtSlot, QSettings, Qt
+from PyQt4.QtGui import (
+    QApplication, QCheckBox, QDesktopServices, QDialog, QFont, QHBoxLayout,
+    QLabel, QMessageBox, QPixmap, QPushButton, QStyle, QTextBrowser,
+    QVBoxLayout, QWidget
+)
+
+try:
+    from Orange.widgets.widget import OWWidget
+    from Orange.version import full_version as VERSION_STR
+except ImportError:
+    # OWWidget (etc.) is not available because this is not Orange
+    class OWWidget: pass
+    VERSION_STR = '???'
+
+
+REPORT_POST_URL = 'http://qa.orange.biolab.si/error-report/v1/'
+ISSUE_TRACKER_URL = 'https://github.com/biolab/orange3'
+
+THUMBSUP_IMAGE = os.path.join(os.path.dirname(__file__), 'thumbsup.png')
+
+log = logging.getLogger()
+
+
+class ErrorReporting(QDialog):
+    _cache = set()  # For errors already handled during one session
+
+    class DataField:
+        EXCEPTION = 'Exception'
+        MODULE = 'Module'
+        WIDGET_NAME = 'Widget Name'
+        WIDGET_MODULE = 'Widget Module'
+        VERSION = 'Version'
+        ENVIRONMENT = 'Environment'
+        MACHINE_ID = 'Machine ID'
+        WIDGET_SCHEME = 'Widget Scheme'
+        STACK_TRACE = 'Stack Trace'
+
+    def __init__(self, data):
+        icon = QApplication.style().standardIcon(QStyle.SP_MessageBoxCritical)
```

Comment:
The comments timeline doesn't show for it, but the icon has been changed into warning.